src/sdl/window.cpp

```diff
@@ -15,24 +15,22 @@
 
 #include "sdl/window.hpp"
 
+#include "preferences/preferences.hpp"
 #include "sdl/exception.hpp"
 #include "sdl/surface.hpp"
 
-#include <SDL2/SDL_render.h>
+#include <SDL3/SDL_render.h>
 
 namespace sdl
 {
 
 window::window(const std::string& title,
-				 const int x,
-				 const int y,
 				 const int w,
 				 const int h,
-				 const uint32_t window_flags,
-				 const uint32_t render_flags)
-	: window_(SDL_CreateWindow(
-		title.c_str(), x, y, w, h, window_flags | SDL_WINDOW_HIDDEN))
-	, pixel_format_(SDL_PIXELFORMAT_UNKNOWN)
+				 const uint32_t window_flags)
+	: window_(SDL_CreateWindow(title.c_str(), w, h, window_flags | SDL_WINDOW_HIDDEN))
+	// TODO SDL3: SDL doesn't allow querying the renderer's pixel format anymore, is this fine?
```

Comment:
Just remove `pixel_format_` and its getters from this class entirely. The only places it's used is as an argument to `SDL_CreateTexture` in video.cpp, which can be simplified anyway.

Reply:
updated.